### PR TITLE
User guide and Developer guide for Orchestration

### PR DESCRIPTION
**Pull request by kumarashit**
Adding user guide for Orchestration. It contains steps to install and use the services

### CODE REVIEW

File: readthedocs/starterguides/userguides/Orchestration_UserGuide.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Click on the 'instacnes' button on 'Services' page to get the list of all instances of the services being created and its status.
+
+## Known issues
+### Multiple service registration in a loop may fail sometime
+Workaround: If service registration is done through CURL/Postman from a script, put a delay of 60 secs between every service registration
+
+### Others:
+Please refer to https://github.com/opensds/orchestration/blob/master/docs/INSTALL.md for some of the known issues while installting some dependent components.
+
+## References
```
Comment (joseph-v) [inline readthedocs/starterguides/userguides/Orchestration_UserGuide.md]: You may add reference to Developer Guide here

File: readthedocs/starterguides/userguides/Orchestration_UserGuide.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            }
+        ]
+    }
+
+Please note that the wfe_type should be 'st2' as OpenSDS supports only 'StackStorm' as Orchestration Manager currently. You can specify any existing group like 'Provisioning' or 'Migration' or create a new group altogether.
+
+Once these steps are done, user can go into the OpenSDS dashboard and create and execute an instance.
+
+### Checking created instacnes
+Click on the 'instacnes' button on 'Services' page to get the list of all instances of the services being created and its status.
```
Comment (joseph-v) [inline readthedocs/starterguides/userguides/Orchestration_UserGuide.md]: typo *instacnes*

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+            'content-type': 'application/json',
+            'x-auth-token': auth_token
+        }
+        r = requests.post(url=url, data=json.dumps(data), headers=headers)
+        r.raise_for_status()
+        resp = r.json()
+        return resp["id"]
+```
+### Points to remember:
+    a) Every python script need to import 'Action' from st2common.eunners.base_action
```
Comment (joseph-v) [inline readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: type eunners -> runners

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+To verify you can login into the StackStorm UI and execute the workflow by provising the required inputs.
+Alternatively, you can use StackStorm CLI to verify:
+## St2 command to verify
+    $ st2 action execute opensds.<Your workflow name>  <All input parameters separated by space>
+
+# Adding Services into OpenSDS
+OpenSDS has different Services for Orchestrating the tasks. These services are grouped under a category. Ex: provision-volume Service is grouped under 'Provisioning'. 
+Each service consists of workflow. Like the provision-volume service under the 'Provisioning' group consist of 'opensds.provision-volume'. bucket-migration service under the 'Migration' group consist of 'opensds.bucket-migration' workflow
+Developer can define their own 'Group' and Service name.
+To Add a service for a workflow in OpenSDS, orchestration cli 'orchctl'
```
Comment (joseph-v) [inline readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: reword something like - "may use Orchestration REST calls or Orchestration CLI 'orchctl' "

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md [hunk 5]
```diff
@@ -0,0 +1,364 @@
+# Developer Guide: Orchestration and automation
+# Overview
+This guide is intended for Orchstration and Automation developers. It provides detailed steps to write new workflows for sequence of actions and Orchestrate them through OpenSDS. To undestand the feature and requirements in detail, please refer to the Orchestration user guide and Orchestration design doc. OpenSDS currently uses StackStorm as the Orchestration manager. It uses Mistral workflows of StackStorm.
```
Comment (nguptaopensds) [inline readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: Orchstration---> Orchestration

**Comment by xing-yang** [comment]
In the document, please also clarify that user/developer needs to log into the stackstorm container to find /opt/stackstorm/... if installed using docker image.

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md (hunk 5 above)
Comment (kumarashit) [reply readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: Done

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md (hunk 4 above)
Comment (kumarashit) [reply readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: Done

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md (hunk 3 above)
Comment (kumarashit) [reply readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: Done

File: readthedocs/starterguides/userguides/Orchestration_UserGuide.md (hunk 2 above)
Comment (kumarashit) [reply readthedocs/starterguides/userguides/Orchestration_UserGuide.md]: Done

File: readthedocs/starterguides/userguides/Orchestration_UserGuide.md (hunk 1 above)
Comment (kumarashit) [reply readthedocs/starterguides/userguides/Orchestration_UserGuide.md]: Have added as TODO. Once Developer guide is merged, will add link

File: readthedocs/starterguides/userguides/Orchestration_UserGuide.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
+# Opensds Orchestration
+
+## Introduction
+OpenSDS is aimed at addressing the storage integration challenges of both the Cloud Native environmnet and traditional IT environment.In a distributed environment, the business processes consist of several interconnected steps which may require streamline the process. OpenSDS provides orchestrating these process to ensure accurate and faster execution with minimal manual intervention. The Orchestration framework provides flexibility to use existing workflows or define customized workflows to get the simplied execution of tasks.
```
Comment (xing-yang) [inline readthedocs/starterguides/userguides/Orchestration_UserGuide.md]: s/environmnet/environment

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+    a) Every python script need to import 'Action' from st2common.eunners.base_action
+    b) Define your action class which inherits from 'Action' base class
+    c) This class must define the 'run' function
+    d) All the business logic to execute the action goes into this 'run' method
+    e) Parameters for the 'run' method is the class object ('self' here) and all the required parameters for this action. These parameters are same that was defined in the corresponding action YAML
+
+## How to load it into the Orchestration Manager
+Once the workflows, actions and runners are defined, these workflows need to be registered into the Orchestration Manager, StackStorm.
+If the StackStorm is installed using Docker image:
+Go to the interactive bash cell of the StackStorm docker instance and follow the steps below
```
Comment (xing-yang) [inline readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: s/cell/shell

File: readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+Else
+Go to the command prompt and execute:
```
Comment (xing-yang) [inline readthedocs/starterguides/developerguides/Orchestration_Developer_Guide.md]: Please combine lines 199-202 into one paragraph.  The newlines won't have any effect anyway in the resulting doc.

**Comment by xing-yang** [comment]
@kumarashit I need to get this merged.  Please address comments with a follow-up PR.  Thanks.

**Pull request merged**